src/components/Header/ChangeRolesView/index.js

```diff
@@ -39,109 +15,93 @@ function ChangeRole({ isToggle, role, setRoleView, roleView, uid }) {
-    localStorage.getItem(ID_TO_SELECTED_ROLE_MAP_KEY)
-  )?.[uid]; // || localStorage.getItem(SELECTED_ROLE_KEY);
-  const hasLastSelectedRole = roles.some(
-    (role) => role.key === lastSelectedRoleKey
-  );
-  const [roleView, setRoleView] = React.useState(
-    hasLastSelectedRole ? lastSelectedRoleKey : defaultRole?.key
-  );
-  const otherRole =
-    roles[(roles.findIndex((role) => role.key === roleView) + 1) % 2];
-
-  const commonProps = { setRoleView, roleView, uid };
-  const history = useHistory();
-  let menu = "";
+  const [roleView, setRoleView] = useState(null);
+  let menu = null;
 
-  if (roles.length > 2) {
-    const menuContents = (
-      <div>
-        {roles.map((role) => (
-          <ChangeRole role={role} key={role.key} {...commonProps} />
-        ))}
-      </div>
+  useEffect(() => {
+    const defaultRole = roles.find((role) => role.assignedRole) || roles[0];
+    const lastSelectedRoleKey = JSON.parse(
+      localStorage.getItem(ID_TO_SELECTED_ROLE_MAP_KEY)
+    )?.[uid];
+    const hasLastSelectedRole = roles.some(
+      (role) => role.key === lastSelectedRoleKey
     );
-    menu = leftDrawer ? (
-      <AccordionDropDownMenu textMsgKey="SWITCH_VIEWS">
-        {menuContents}
-      </AccordionDropDownMenu>
-    ) : (
-      <TextButtonDropDownMenu
-        btnTextMsgKey="SWITCH_VIEWS"
-        attachRight={!leftDrawer}
-        menuContainerProps={{
-          // sx: { mt: '45px' },
-          id: "menu-appbar",
-          /*
-            anchorOrigin: {
-              vertical: "top",
-              horizontal: leftDrawer ? "left" : "right",
-            },
-            keepMounted: true,
-            transformOrigin: {
-              vertical: "top",
-              horizontal: leftDrawer ? "left" : "right",
-            },
-            */
-        }}
-      >
-        {menuContents}
-      </TextButtonDropDownMenu>
-    );
-  } else if (roles.length === 2) {
-    menu = <ChangeRole isToggle={true} role={otherRole} {...commonProps} />;
-  }
-  //const location = useLocation();
+    setRoleView(hasLastSelectedRole ? lastSelectedRoleKey : defaultRole?.key);
+  }, [roles, uid]);
 
-  React.useEffect(() => {
-    setRole(roleView);
-    // if (location !== roles[roleView]) {
-    // for load
-    //  history.push(roles[roleView]);
-    //}
+  useEffect(() => {
+    if (roleView) {
+      setRole(roleView);
+    }
   }, [roleView]);
 
-  // React.useEffect(() => {
-  //   const roleData = roles.find((role) => role.key === roleView);
-  //   roleData?.landingPage && history.push(roleData.landingPage);
-  // }, []);
+  const handleRoleChange = (selectedRole) => {
+    setRoleView(selectedRole);
+  };
+
+  if (roles.length > 1) {
+    const menuContents = roles.map((role) => (
+      <ChangeRole
+        role={role}
+        key={role.key}
+        isToggle={roles.length > 2}
+        setRoleView={handleRoleChange}
+        roleView={roleView}
+        uid={uid}
+      />
+    ));
+
+    if (roles.length > 2) {
```

Comment:
We also need to check that `leftDrawer` is `true` (meaning the mobile sidebar is open) or we'll get something like this:
![image](https://github.com/navgurukul/bhanwari-devi/assets/9159008/d28bc9dc-e6ad-4c07-8cd7-1dc047f43c6c)
```suggestion
    if (roles.length > 2 && leftDrawer) {
```